src/__tests__/index.test.ts

```diff
@@ -133,6 +133,19 @@ describe("shallowEqual on objects", () => {
   });
 });
 
+describe("shallowEqual on mixed array/objects", () => {
+  it("should act correctly when types are mixed", () => {
+    expect(shallowEqual(["boris", "johnson"], { boris: "johnson" })).toBe(
+      false
+    );
+    expect(shallowEqual({ boris: "johnson" }, ["boris", "johnson"])).toBe(
+      false
+    );
+    expect(shallowEqual(["boris", "johnson"], { liz: "truss" })).toBe(false);
+    expect(shallowEqual({}, [])).toBe(false);
```

Comment:
`shallowEqualObjects` actually says these are equal (in its current implementation). We could correct that implementation by adding `isArray` checks — but LMK how you feel about that, since the caller is explicitly saying, by calling it, that they know they're comparing objects.